osu.Game/Screens/Edit/Compose/Components/SelectionBoxDragHandleContainer.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Screens.Edit.Compose.Components
+{
+    /// <summary>
+    /// Represents a display composite containing and managing the visibility state of the selection box's drag handles.
+    /// </summary>
+    public class SelectionBoxDragHandleContainer : CompositeDrawable
```

Comment:
Reading all the intertwined logic you have setup in this (and related) classes makes me sleepy.

Also, it's halfway between being flexible and not. For instance, what if we decide the bottom buttons should only display when the bottom centre drag handle is being hovered?

I don't have a solution in mind; just commenting that this is a pretty convoluted setup.